src/main/java/gregtech/integration/jei/multiblock/MultiblockInfoRecipeWrapper.java

```diff
@@ -307,11 +310,36 @@ public boolean handleClick(Minecraft minecraft, int mouseX, int mouseY, int mous
                     tooltip.set(k, TextFormatting.GRAY + tooltip.get(k));
                 }
             }
+            MetaTileEntity mte = GregTechAPI.META_TILE_ENTITY_REGISTRY.getObjectById(tooltipBlockStack.getItemDamage());
+            if(mte instanceof IMultiblockAbilityPart) {
+                MultiblockAbility ability = ((IMultiblockAbilityPart) mte).getAbility();
+                Map<MultiblockAbility, List<String>> abilityTooltipMap = infoPage.getAbilityTooltipMap();
+                if(abilityTooltipMap.containsKey(ability)) {
+                    List<String> tooltips = abilityTooltipMap.get(ability);
+                    for(int i = 0; i < tooltips.size(); i++) {
+                        tooltip.add(i + 1, TextFormatting.GREEN + I18n.format(tooltips.get(i)));
+                    }
+                }
+            }
             return tooltip;
         }
         return Collections.emptyList();
     }
 
+    public void addAbilityPartTooltip(int slotIndex, boolean input, ItemStack itemStack, List<String> tooltip) {
+        MetaTileEntity mte = GregTechAPI.META_TILE_ENTITY_REGISTRY.getObjectById(itemStack.getItemDamage());
+        if(mte instanceof IMultiblockAbilityPart) {
+            MultiblockAbility ability = ((IMultiblockAbilityPart) mte).getAbility();
+            Map<MultiblockAbility, List<String>> abilityTooltipMap = infoPage.getAbilityTooltipMap();
+            if(abilityTooltipMap.containsKey(ability)) {
+                List<String> tooltips = abilityTooltipMap.get(ability);
+                for(int i = 0; i < tooltips.size(); i++) {
+                    tooltip.add(i + 1, TextFormatting.GREEN + I18n.format(tooltips.get(i)));
```

Comment:
All tooltips added this way will be green tooltips it could be interesting to allow other colors and formating or at least not enforce green color.